### DISCUSSION TITLE
Charging ETH fees instead of stablecoin fees

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/flow#327] [Ideas]
We've recently decided to charge [ETH fees](https://github.com/sablier-labs/v2-core/issues/1076) instead of [protocol fees](https://github.com/sablier-labs/v2-core/issues/1034) in Lockup. ETH fees make much more sense for Lockup streams per all the reasons provided [here](https://github.com/sablier-labs/company-discussions/discussions/72#discussioncomment-11115807).

Now, for Flow streams, the fee is paid in stablecoin. This is not bad, but I wonder if charging in ETH wouldn't be better?

1. More consistency across our fee models
2. Less psychological resistance; the user has to pay a fee in ETH anyway (for the gas), so charging $1 in ETH should feel less pressuring than paying a *separate* fee in USDC (from the recipient's salary)
3. Charging in ETH means that we are charging the account that withdraws, not the recipient (these accounts might not be the same)

Notes:

- I'm not suggesting to change the implementation now because doing so would cancel the validity of the security audits.
- We will start charging in ~March anyway, so changing the implementation in Flow now would be overkill.

Cc @sablier-labs/everybody for feedback.

**Comment by maxdesalle** [comment]
We should 100% charge in ETH as opposed to in stablecoins. As you said, it leads to consistency across our products, and it would be confusing for users if for some products they have to pay in ETH, and for others they need stablecoins. And the decreased psychological resistance you mentioned is very true as well.

I think an additional argument in favor of charging in ETH as opposed to in stablecoins, and this goes along with your second point, is that it makes for a significantly better user experience as people have ETH in their wallet anyway.

**Reply by PaulRBerg** [reply]
Wonderful. Couldn't agree more.

Onwards to a great 2025 for Sablier.

I will tentatively open a GitHub issue for this so we can get it implemented hopefully before December 16 (when the next audit starts).

**Reply by andreivladbrg** [reply]
> I will tentatively open a GitHub issue for this so we can get it implemented hopefully before December 16 (when the next audit starts).

this is flow discussion and you said: "I'm not suggesting to change the implementation now because doing so would cancel the validity of the security audits."

**Reply by PaulRBerg** [reply]
We can develop and ship Flow 1.1.0 until December, and ask the auditors to audit Flow 1.1.0

**Reply by andreivladbrg** [reply]
> We can develop and ship Flow 1.1.0 until December

hmm, but in that case there will be no 1.0.0 as it didn't go live :))

**Reply by PaulRBerg** [reply]
We will ship 1.0.0 in December and start the development and audit for 1.1.0 in parallel, which we will then ship in March.

**Reply by andreivladbrg** [reply]
ok got it, but that's not you said in your above comments

**Reply by PaulRBerg** [reply]
Yes. What I said above is to get it implemented before the audit starts.

I don't understand where the confusion/ problem is?

**Comment by andreivladbrg** [comment]
Same here, I am 100% in favor of having ETH fees instead of 'streamed' token fees

**Comment by razgraf** [comment]
+1

**Comment by smol-ninja** [comment]
Always in favour of ETH fees.

**Comment by smol-ninja** [comment]
Closing this discussion as resolved. Related issue: https://github.com/sablier-labs/flow/issues/328
